src/main/java/src/ProductList.java

```diff
@@ -2,20 +2,91 @@
 
 import java.util.ArrayList;
 
+import static src.Product.*;
+
 public class ProductList {
-    private final String name;
-    private ArrayList<Product> listOfProducts;
+  private final String name;
+  private final ArrayList<Product> listOfProducts;
+
+  public ProductList(String name, ArrayList<Product> listOfProducts) {
+    this.name = sanitizeString(name);
+    this.listOfProducts = listOfProducts;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  private static String sanitizeString(String name) {
+    return name.replace("\n", "").replace("$$", "").replace("}}", "");
+  }
+
+  public ArrayList<Product> getListOfProducts() {
+    return listOfProducts;
+  }
 
-    public ProductList(String name, ArrayList<Product> listOfProducts) {
-        this.name = name;
-        this.listOfProducts = new ArrayList<>();
+  public static String serializeProductList(ProductList productList) {
+    if (productList == null) {
+      throw new IllegalArgumentException("ProductList of argument is null");
     }
+    return "ProductList{"
+        + "\n$$name$$"
+        + "#"
+        + productList.getName()
+        + "#"
+        + ",\n$$listOfProducts$$"
+        + "#"
+        + Product.serializeListOfProducts(productList.listOfProducts)
+        + "#"
+        + "}}";
+  }
 
-    public String getName() {
-        return name;
+  public static String serialize(ArrayList<ProductList> productLists) {
+    if (productLists == null) {
+      throw new IllegalArgumentException("ProductList Array of  argument is null");
     }
+    String[] stringList = new String[productLists.size()];
+    StringBuilder productString = new StringBuilder();
+    for (int i = 0; i < productLists.size(); i++) {
+      stringList[i] = serializeProductList(productLists.get(i));
+      productString.append(stringList[i]);
+    }
+    return productString.toString();
+  }
+
+  public static ArrayList<Product> deserialize(String productString) {
```

Comment:
This seems like duplicated code.

Reply:
This method is used to serialize a list of ProductList.